RedHatInsights/insights-remediations-frontend
# fix(detailsTable): bulk select fix

# Description

Associated Jira ticket: # (issue)
Fixing bulk select in Actions and Systems tab
https://issues.redhat.com/browse/REMEDY-283
https://issues.redhat.com/browse/REMEDY-145
https://issues.redhat.com/browse/REMEDY-90

Please include a summary of the change, what this fixes/creates/improves.
I added 2 buttons:
**Select page (number of rows)
Select all (number of values in the entire table)**
I improved the behavior of the select page buttons

**Behavior**: User has a list of things, 10 per page, 5 pages (50 items total):
First page - “Select page” --> 10 items selected
Nav to second page - “Select page” --> 20 items selected (adds to the selection)
Nav to third page - "Select page" --> 30 items selected (adds to the selection)
To unselect the third page -> "Select page" --> 20 items selected (takes out 10 - the page - from the selection)
Nav to the fourth page - “Select all” --> 50 items selected (all selected)
User chooses “Select none” --> Everything deselected, 0 items selected
User selects a single item --> 1 item selected
User goes back to page 2, chooses “Select page” --> page 2 is selected & that one item from page, 11 items total


This is a diagram for the select page button in Systems tab so you could understand it better
![image](https://user-images.githubusercontent.com/62722417/213154476-362d7f81-5f59-42ad-8956-653b76e0ad61.png)

This is a diagram for the select page button in Actions tab
![image](https://user-images.githubusercontent.com/62722417/213429985-afc05c08-f0f2-4a77-9ad6-ed3bf685d3af.png)

# How to test the PR
Find remediation with the name "voznesenskii test" to test systems tab.
Find remediation with the name "test-patch" to test actions tab.
Open the local dev environment, go to the /insights/remediations and find remediation with several systems
Open the table and check bulk select in the Actions tab and Activity tab.

What to check:
The dropdown should be disabled when there are no systems/actions to show. 
The dropdown should be enabled when there are systems/actions to show.
If you select the number of values < than the whole table the check mark should look like a - symbol
![image](https://user-images.githubusercontent.com/62722417/212057467-665175f7-eef0-4c2e-9419-96a7462f462c.png)

If you select the number of values = to the amount of values in the entire table the check mark should look like this 
![image](https://user-images.githubusercontent.com/62722417/212065791-6016a301-b9d2-4844-a1d5-d53bd021b6d9.png)

# Before the change
There were no select all button

# After the change
![image](https://user-images.githubusercontent.com/62722417/212066852-9e34c49e-f914-4166-b880-059df02021de.png)




# Dependent work link


# Checklist:
